src/main/java/Duke.java

```diff
@@ -1,10 +1,409 @@
+     *
+     * @param todoDescription Todo description.
+     * @return Feedback display message for adding a new todo.
+     */
+    public static String executeAddTodo(String todoDescription) {
+        // Create a new Todo instance
+        Todo todo = new Todo(todoDescription);
+
+        return executeAddTask(todo);
+    }
+
+    /**
+     * Returns a message when user's command keyword does not belong to any valid keywords.
+     *
+     * @return Invalid input message.
+     */
+    public static String displayMessageForInvalidInput() {
+        return MESSAGE_FOR_INVALID_INPUT;
+    }
+
+    /**
+     * Adds a new task to tasks array.
+     *
+     * @param task Task object to be added.
+     * @return Feedback display message for adding a new task.
+     */
+    public static String executeAddTask(Task task) {
+        tasks[taskCount] = task;
+        taskCount++;
+
+        return String.format(HORIZONTAL_LINE + LS
+                + MESSAGE_ADD_TITLE + LS
+                + task.toString() + LS
+                + MESSAGE_ADD_CONCLUSION + System.lineSeparator()
+                + HORIZONTAL_LINE + System.lineSeparator(), taskCount);
+    }
+
+    /**
+     * Lists out tasks added so far with their status.
+     */
+    public static String executeListAllTasks() {
+        String feedback = HORIZONTAL_LINE + LS
+                + MESSAGE_LIST_TITLE + System.lineSeparator();
+
+        // Iterate through tasks array and print each task with its status and description
+        for (int i = 0; i < taskCount; i++) {
+            feedback += LINE_PREFIX + (i + 1) + SEPARATOR_TASK_NUMBER_TASK_DESC
+                    + tasks[i].toString() + System.lineSeparator();
+        }
+
+        feedback += HORIZONTAL_LINE + System.lineSeparator();
+        return feedback;
+    }
+
+    /**
+     * Marks a task in the tasks array as done.
+     *
+     * @param commandArgs Full command args string from the user.
+     * @return Feedback display message for marking the task as done.
+     */
+    public static String executeMarkTaskAsDone(String commandArgs) {
+        int taskIndex = extractTaskIndexFromInputString(commandArgs);
+
+        // Update status of task
+        tasks[taskIndex].markAsDone();
+
+        return HORIZONTAL_LINE + LS
+                + MESSAGE_DONE_TITLE + LS
+                + tasks[taskIndex].toString() + System.lineSeparator()
+                + HORIZONTAL_LINE + System.lineSeparator();
+    }
+
+    /**
+     * Converts task number in user's command (starting from 1)
+     * to the corresponding task index  in tasks list (starting from 0).
+     * In the case of "done X" command,
+     * the command argument X is the task number to be marked as done.
+     *
+     * @param commandArgs User's argument passed in the command.
+     * @return Task index.
+     */
+    public static int extractTaskIndexFromInputString(String commandArgs) {
+        return Integer.parseInt(commandArgs) - 1;
+    }
+
```

Comment:
Meticulous commenting, and proper indentations and spacing. No coding violations, very high readability. Nice!!

Reply:
Thank you for your comments :D